sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,281 @@
+    is_solvable = True
+
+    def __init__(self, pcgs_):
+        self.perm_group = PermutationGroup(pcgs_)
+        self.pc_series = self._pc_series()
+        self.pcgs = self._compute_pcgs()
+
+    def _pc_series(self):
+        return self.perm_group.composition_series()
+
+    def _compute_pcgs(self):
+        # computes the generating sequence for polycyclic groups.
+        series = self.pc_series
+        pcgs = []
+        for i in range(len(series)-1):
+            for g in series[i].generators:
+                if not g in series[i+1]:
+                    pcgs.append(g)
+        return pcgs
+
+    def relative_orders(self):
+        rel_orders = []
+        for i in range(len(self.pc_series)-1):
+            G = self.pc_series[i]
+            H = self.pc_series[i+1]
+            rel_orders.append(G.order()//H.order())
+        return rel_orders
+
+    def is_prime_order(self):
+        for order in self.relative_orders():
+            if order not in sieve:
+                return False
+        return True
+
+    def length(self):
+        return len(self.pcgs)
+
+    def pc_element_exponent(self, element):
+        series = self.pc_series
+        pcgs = self.pcgs
+        exponent = [0]*len(series)
+        for i in range(len(series)):
+            exp = 0
+            if not element in series[i]:
+                for j in range(len(pcgs)):
+                    element = (pcgs[j]**-1)*element
+                    exp = exp + 1
+                    if element in series[i]:
+                        exponent[i] = exp
+                        break
+        return exponent
+
+
+class Collector(DefaultPrinting):
+
+    """
+    References
+    ==========
+
+    .. [1] Holt, D., Eick, B., O'Brien, E.
+           "Handbook of Computational Group Theory"
+           Section 8.1.3
+    """
+
+    def __init__(self, pc_relators, relative_order, group):
+        self.pc_relators = pc_relators
+        self.relative_order = relative_order
+        self.group = group
+
+    def minimal_uncollected_subword(self, word):
+        """
+        Returns the minimal uncollected subwords.
+
+        Examples
+        ========
+        >>> from sympy.combinatorics.pc_groups import Collector
+        >>> from sympy.combinatorics.free_groups import free_group
+
+        Example 8.7 Pg. 281 from [1]
+        >>> F, x1, x2 = free_group("x1, x2")
+        >>> pc_relators = {x1**2 : (), x1*x2*x1**-1 : x2**-1, x1**-1*x2*x1 : x2**-1}
+        >>> relative_order = {x1: 2}
+        >>> word = x2**2*x1**7
+        >>> group = word.group
+        >>> collector = Collector(pc_relators, relative_order, group)
+        >>> collector.minimal_uncollected_subword(word)
+        ((x2, 2), (x1, 1))
+
+        """
+        # To handle the case word = <identity>
+        if not word:
+            return None
+
+        group = self.group
+        index = {s: i+1 for i, s in enumerate(group.symbols)}
+        array = word.array_form
+        re = self.relative_order
```

Comment:
Could this be a list? For instance, `[2, 3, 2, 2]` for S(4).

Reply:
Maybe `None` could be used to denote a missing power relation ("infinite relative order").

Reply:
Yes, we can do that!